Circus dies with exception when using sockets combined with hooks

This is my config:

``` ini
[watcher:test]
copy_env = true
cmd = /usr/bin/python
args = -m longrunning.module
numprocesses = 1
singleton = true
graceful_timeout = 1
hooks.before_signal = longrunning.hooks:graceful_dead

[socket:mhost-socks5]
host = 0.0.0.0
port = 45051
```

It makes no difference when using UNIX sockets.
The hook is a function that just returns True or False.
This is the exception I get when restarting circus completely or calling `circusctl restart`

```
2014-12-04 21:05:27 circus[13147] [ERROR] exception unbound method close() must be called with socket instance as first argument (got CircusSocket instance instead) caught
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 227, in wrapper
    runner.run()
  File "/usr/local/lib/python2.7/dist-packages/tornado/gen.py", line 531, in run
    yielded = self.gen.send(next)
  File "/usr/local/lib/python2.7/dist-packages/circus/arbiter.py", line 532, in start
    self.stop_controller_and_close_sockets()
  File "/usr/local/lib/python2.7/dist-packages/circus/arbiter.py", line 540, in stop_controller_and_close_sockets
    self.sockets.close_all()
  File "/usr/local/lib/python2.7/dist-packages/circus/sockets.py", line 199, in close_all
    sock.close()
  File "/usr/local/lib/python2.7/dist-packages/circus/sockets.py", line 94, in close
    socket.socket.close(self)
```

--- a/circus/sockets.py
+++ b/circus/sockets.py
@@ -176,7 +176,7 @@ class CircusSocket(socket.socket):
         return 'socket %r at %s' % (self.name, self.location)
 
     def close(self):
-        socket.socket.close(self)
+        super(CircusSocket, self).close()
         if self.is_unix and os.path.exists(self.path):
             os.remove(self.path)
 

